pt/lessons/advanced/metaprogramming.md

```diff
@@ -0,0 +1,246 @@
+---
+layout: page
+title: Metaprogramação
+category: advanced
+order: 7
+lang: pt
+---
+
+Metaprogramação é o processo de utilização de código para escrever código. Em Elixir isso nos dá a capacidade de estender a linguagem para se adequar às nossas necessidades e dinamicamente alterar o código. Vamos começar observando como Elixir é representado por debaixo dos panos, em seguida como modificá-lo, e finalmente, como podemos usar esse conhecimento para estendê-lo.
+
+
+Uma palavra de cautela: Metaprogramação é complicado e só deve ser usado quando for absolutamente necessário. O uso excessivo certamente levará a um código complexo que é difícil de entender e debugar.
+
+## Sumário
+
+- [Quote](#quote)
+- [Unquote](#unquote)
+- [Macros](#macros)
+	- [Macros Privados](#macros-privados)
```

Comment:
is this indentation level correct? I think it could be 2 spaces inside the upper level.